Analyzer/plugins/Analyzer.cc

```diff
@@ -1109,8 +1148,35 @@ void Analyzer::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup)
             if(id == 130) track_PFIso05_sumNeutHadPt+=pt;
             if(id == 22) track_PFIso05_sumPhotonPt+=pt;
         }
+
+        float drForMiniIso = 0.0;
+        if (track->pt() < 50 ) {
+          drForMiniIso = 0.2;
+        } else if (track->pt() < 200) {
+          drForMiniIso = 10/track->pt();
+        } else {
+          drForMiniIso = 0.05;
+        }
+        if (dr<drForMiniIso) {
+            // charged cands from PV get added to trackIso
+            if(pf_isChHadronForIdx && fromPV) track_PFMiniIso_sumCharHadPt+=pt;
+            // charged cands not from PV get added to pileup iso
+            else if(pf_isChHadronForIdx) track_PFMiniIso_sumPUPt+=pt;
+            // neutral hadron iso
+            if(pf_isNeutHadronForIdx) track_PFMiniIso_sumNeutHadPt+=pt;
+            // photon iso
+            if(pf_isPhotonForIdx) track_PFMiniIso_sumPhotonPt+=pt;
+            // muon iso
+            if(pf_isMuonForIdx) track_PFMiniIso_sumMuonPt+=pt;
+        }
       }//end loop PFCandidates
     }
+    
+  // Calculate PF mini relative isolation
+  float miniRelIsoAll = (track_PFMiniIso_sumCharHadPt + std::max(0.0, track_PFMiniIso_sumNeutHadPt + track_PFMiniIso_sumPhotonPt - 0.5* track_PFMiniIso_sumPUPt))/track->pt();
+  float miniRelIsoAll_wMuon = (track_PFMiniIso_sumMuonPt + track_PFMiniIso_sumCharHadPt + std::max(0.0, track_PFMiniIso_sumNeutHadPt + track_PFMiniIso_sumPhotonPt - 0.5* track_PFMiniIso_sumPUPt))/track->pt();
```

Comment:
what about the electron?

Reply:
right but what's the motivation to not include them? maybe something to discuss tomorrow